common/web/gesture-recognizer/src/engine/headless/gestures/matchers/matcherSelector.ts

```diff
@@ -0,0 +1,325 @@
+     * When we iterate through on update-steps, we go sequentially; the first Promise to be marked
+     * 'resolved' wins.
+     */
+    this.potentialMatchers.sort((a, b) => b.model.resolutionPriority - a.model.resolutionPriority);
+
+    // Now that all GestureMatchers are built, reset ALL of our sync-update-check hooks.
+    this.resetSourceHooks();
+
+    return matchPromise.corePromise;
+  }
+
+  private readonly attemptSynchronousUpdate = () => {
+    const sourceCurrentTimestamps = this._sourceSelector.map((tracker) => tracker.source.isPathComplete ? null : tracker.source.currentSample.t);
+    const t = sourceCurrentTimestamps[0];
+
+    // Ignore timestamps from already-terminated paths; they should not block synchronicity checks.
+    if(sourceCurrentTimestamps.find((t2) => (t2 !== null) && (t != t2))) {
+      return;
+    }
+
+    this.potentialMatchers.forEach((matcher) => matcher.update());
+  };
+
+  private resetSourceHooks() {
+    const resetHooks = (gestureSource: GestureSourceSubview<Type>) => {
+      // GestureSourceSubviews stay synchronized with their 'base' via event handlers.
+      // We want GestureMatchers to receive all updates before we attempt a sync'd update.
+      const baseSource = gestureSource.baseSource;
+
+      // So, a resetHooks call says to remove the old handler...
+      baseSource.path.off('step', this.attemptSynchronousUpdate);
+      baseSource.path.off('complete', this.attemptSynchronousUpdate);
+      baseSource.path.off('invalidated', this.attemptSynchronousUpdate);
+
+      // And re-add it, but at the end of the handler list.
+      baseSource.path.on('step', this.attemptSynchronousUpdate);
+      baseSource.path.on('complete', this.attemptSynchronousUpdate);
+      baseSource.path.on('invalidated', this.attemptSynchronousUpdate);
+    }
+
+    // Make sure our source-watching hooks are the last handler for the event;
+    // matcher-handlers should go first.  (Due to how subview synchronization works)
+    this._sourceSelector.forEach((entry) => resetHooks(entry.source));
+  }
+
+  private matchersForSource(source: GestureSource<Type>) {
+    return this.potentialMatchers.filter((matcher) => {
+      return !!matcher.pathMatchers.find((pathMatch) => pathMatch.source == source)
+    });
+  }
+
+  private matcherSelectionFilter(matcher: GestureMatcher<Type>, matchSynchronizers: ManagedPromise<any>[]) {
+    // Returns a closure-captured Promise-resolution handler used by individual GestureMatchers managed
+    // by this class instance.
+    return async (result: MatchResult<Type>) => {
+      // Note:  is only called by GestureMatcher Promises that are resolving.
+
+      // Ensure that any essentially-synchronous resolving GestureMatchers resolve in order of
+      // their specified `resolutionPriority`, with larger values first.
+      await this.promisePrioritizer.queueWithPriority(matcher.model.resolutionPriority);
```

Comment:
Huh.  Revisited this after noting the main comment of https://github.com/keymanapp/keyman/pull/9453#pullrequestreview-1573201164:

> I don't really understand the big picture gestures use case as the mechanical unit test of correctness doesn't really explain why the gestures design would need such a thing.

I did have a vacation since then, so I figured the refresher might help.  Turns out... the unit tests actually pass now with this disabled!

I _think_ I see what happened now:

https://github.com/keymanapp/keyman/blob/d03cd7f155313b486f802b5b0be93f54cbebb8ea/common/web/gesture-recognizer/src/engine/headless/gestures/matchers/matcherSelector.ts#L138-L146

I'd been having some serious issues getting certain multitap scenarios to work correctly, largely due to the timing by which source-subview updates were being processed and how that related to the processsing of path-termination.

Lines 145-146 above were part of the last additions to this PR - additions that fixed some "fun" underlying problems with update timings.  Ensuring that updates were processed in the correct order probably fixed the same issues that were previously undermining resolutionPriority ordering - for some gesture types, path updates can trigger resolution of their corresponding gesture matcher, after all.

So... we might actually able able to straight-up drop the Promise prioritizer thing after all.  At least it _did_ help lead me to the true underlying cause for the issues I'd been using it to address.

Reply:
And now... fully dropped.